### PR TITLE
Moved Drag Race Code from IARRC-2017 to Snowflake

### PR DESCRIPTION
**Pull request by garethellis0**
- There's still a fair amount to be done to get this cleaned up and really ready to go, especially around the extra features, like visualization. The obstacle model also leaves much to be desired
- However, everything important is pretty well tested, so I'm OK with merging this. I'll be submitting a issue to get the `TODO`'s cleaned up for next year however.

### CODE REVIEW

File: clang-format/fix_formatting.sh [hunk 1]
```diff
@@ -6,7 +6,15 @@ export CLANG_VERSION=4.0
 # The current directory
 CURR_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-# Run clang-format to see if we need to change anything
-output="$($CURR_DIR/git-clang-format --binary $CURR_DIR/clang-format-$CLANG_VERSION --commit master)"
+# Check that we got exactly one argument
```
Comment (ValRat) [inline clang-format/fix_formatting.sh]: Merge #221 before this :P

File: src/drag_race/include/DragRaceNode.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
+/*
+ * Created By: Gareth Ellis
+ * Created On: July 16th, 2016
+ * Description: An example node that subscribes to a topic publishing strings,
```
Comment (ValRat) [inline src/drag_race/include/DragRaceNode.h]: Description needs updating, else i'll be impressed if we're drag racing with this

File: src/drag_race/src/DragRaceNode.cpp [hunk 3]
```diff
@@ -0,0 +1,183 @@
+/*
+ * Created By: Gareth Ellis
+ * Created On: July 16th, 2016
+ * Description: An example node that subscribes to a topic publishing strings,
```
Comment (ValRat) [inline src/drag_race/src/DragRaceNode.cpp]: TODO here as well

File: src/drag_race/include/DragRaceNode.h (hunk 2 above)
Comment (garethellis0) [reply src/drag_race/include/DragRaceNode.h]: Updated

File: src/drag_race/src/DragRaceNode.cpp (hunk 3 above)
Comment (garethellis0) [reply src/drag_race/src/DragRaceNode.cpp]: Updated

**Comment by garethellis0** [comment]
@ValRat - added a new commit, please re-review

**Pull request merged**